### ISSUE TITLE
Best practice of Express and mongoose

### ISSUE THREAD

**Issue by yashLadha** [iiitv/aua#10]
Need some guidelines for best practices(directory structure, code conventions, etc.) for express and mongoose ODM.

**Comment by aviaryan** [comment]
Please see https://github.com/anotode/anotode-server 
I believe I used express-generator for bootstrapping the project and then customized it.

**Comment by yashLadha** [comment]
Can you share more about what you have done? So that I get a guide, about how to progress?

**Comment by aviaryan** [comment]
Ok. So I first used express-generator to bootstrap the project. But it included some outdated components (like `jade -> pug`) and stuff so I had to manually remove that. 
After that, everything was good to go. 

I recommend going from oldest commit to newer. Commits are atomic and have meaningful names so it should give you a good idea on how to setup an Express-Mongoose project. 
https://github.com/anotode/anotode-server/commits/master?after=1505ab94fe4070733b466695d75d02f87721d178+34
